### PR TITLE
docs: document CMake support

### PR DESCRIPTION
**Pull request by compnerd**
Add some documentation on how to use the CMake based build and some of
the limitations of it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,40 @@ swift build
...
+
+**NOTE**: tests are currently not supported with the CMake based build.
+
+It is highly recommended that you use CMake 3.16 or newer to ensure that `-B` and parallel builds function properly in the example commands below.
+
+macOS:
+
+```
+cmake                                                              \
+  -B /BinaryCache/tensorflow-swift-models                          \
```
Comment (saeta) [inline README.md]: Building into a folder on the root partition is probably a bad idea?

File: README.md [hunk 2]
```diff
@@ -26,6 +26,40 @@ swift build
...
 
+### *Experimental* CMake Support
+
+There is experimental support for building with CMake.  This is required to build the models on Windows, and can also be used to cross-compile the models and the demo programs.
+
+**NOTE**: tests are currently not supported with the CMake based build.
+
+It is highly recommended that you use CMake 3.16 or newer to ensure that `-B` and parallel builds function properly in the example commands below.
+
+macOS:
```
Comment (saeta) [inline README.md]: Does this also work on Linux?

File: README.md [hunk 3]
```diff
@@ -26,6 +26,40 @@ swift build
 swift test
 ```
 
+### *Experimental* CMake Support
+
+There is experimental support for building with CMake.  This is required to build the models on Windows, and can also be used to cross-compile the models and the demo programs.
+
+**NOTE**: tests are currently not supported with the CMake based build.
```
Comment (saeta) [inline README.md]: Note: this is definitely not ideal. :-(

File: README.md (hunk 2 above)
Comment (compnerd) [reply README.md]: Yes, added Linux instructions

File: README.md (hunk 1 above)
Comment (compnerd) [reply README.md]: The docker build does that actually.  It should be obvious which paths to adjust though.  I think having that for the example command is reasonable.

File: README.md (hunk 3 above)
Comment (compnerd) [reply README.md]: Agreed, I hope that we can get that added soon.

**Pull request merged**